### PR TITLE
improve 'GameModePart'

### PR DESCRIPTION
**Pull request by vCamilx**
the use of the 'Stream' API has been removed, also the use of concatenation to convert integers or characters into a string has been changed for better performance (although it is negligible) but also to make the code more readable.

### CODE REVIEW

File: Bukkit/src/main/java/me/fixeddev/commandflow/bukkit/part/GameModePart.java [hunk 1]
```diff
@@ -69,11 +66,16 @@ public GameModePart(String name) {
         }
 
         String possibleGameModeName = next.toUpperCase();
+        List<String> suggestions = new ArrayList<>();
+
+        for (GameMode gameMode : GameMode.values()) {
+            String name = gameMode.name().toLowerCase();
+            if (possibleGameModeName.length() == 0 || name.startsWith(possibleGameModeName)) {
```
Comment (yusshu) [inline Bukkit/src/main/java/me/fixeddev/commandflow/bukkit/part/GameModePart.java:73]: isEmpty()

Comment (vCamilx) [reply Bukkit/src/main/java/me/fixeddev/commandflow/bukkit/part/GameModePart.java]: ready

File: Bukkit/src/main/java/me/fixeddev/commandflow/bukkit/part/GameModePart.java [hunk 2]
```diff
@@ -5,48 +5,45 @@
...
-        GAMEMODE_ALIASES.put(alias.toLowerCase(), gameMode);
-    }
-
     private final String name;
 
     public GameModePart(String name) {
         this.name = name;
     }
 
+    private static void addAlias(String alias, GameMode gameMode) {
```
Comment (yusshu) [inline Bukkit/src/main/java/me/fixeddev/commandflow/bukkit/part/GameModePart.java:43]: Make it `public`

File: Bukkit/src/main/java/me/fixeddev/commandflow/bukkit/part/GameModePart.java (hunk 1 above)
Comment (OcZi) [reply Bukkit/src/main/java/me/fixeddev/commandflow/bukkit/part/GameModePart.java]: `GameMode.values()` could be replaced with already cached values of Map GAMEMODE_ALIASES

File: Bukkit/src/main/java/me/fixeddev/commandflow/bukkit/part/GameModePart.java [hunk 3]
```diff
@@ -69,11 +66,16 @@ public GameModePart(String name) {
         }
 
         String possibleGameModeName = next.toUpperCase();
+        List<String> suggestions = new ArrayList<>();
+
+        for (GameMode gameMode : GameMode.values()) {
```
Comment (yusshu) [inline Bukkit/src/main/java/me/fixeddev/commandflow/bukkit/part/GameModePart.java:71]: GameMode array returned by `GameMode.values()` can be cached, `values()` creates a new array every time it is called

**Review by yusshu — CHANGES_REQUESTED** [review]

File: Bukkit/src/main/java/me/fixeddev/commandflow/bukkit/part/GameModePart.java (hunk 1 above)
Comment (FixedDev) [reply Bukkit/src/main/java/me/fixeddev/commandflow/bukkit/part/GameModePart.java]: Or directly with the aliases

Comment (vCamilx) [reply Bukkit/src/main/java/me/fixeddev/commandflow/bukkit/part/GameModePart.java]: I forgot to say, but I've already changed it

**Pull request merged**